cpp/oneapi/dal/detail/array_utils.hpp

```diff
@@ -0,0 +1,46 @@
+/*******************************************************************************
+* Copyright 2020-2021 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#pragma once
+
+#include "oneapi/dal/array.hpp"
+
+namespace oneapi::dal::detail {
+namespace v1 {
+
+template <typename T>
+class array_via_policy {
```

Comment:
Can you please point where this is used?

Reply:
`homogen_table`, it will be used in other data management classes where we want to have universal code for host and DPC++.